index_configuration_tool/tests/test_main.py

```diff
@@ -0,0 +1,172 @@
+        # Base case should return an empty list
+        self.assertFalse(main.print_report(dict(), dict()))
+        base_data = {
+            "index1": {
+                "settings": {
+                    "bool_key": True,
+                    "num_key": 1
+                },
+                "mappings": {
+                    "str_key": "value"
+                }
+            },
+            "index2": {}
+        }
+        result = main.print_report(base_data, dict())
+        self.assertEqual(2, len(result))
+        self.assertTrue("index1" in result)
+        self.assertTrue("index2" in result)
+        # Index present with exact match
+        test_data = copy.deepcopy(base_data)
+        del test_data["index2"]
+        self.assertFalse(main.print_report(test_data, test_data))
+        # Index has conflict in settings
+        test_data["index1"]["settings"]["num_key"] = -1
+        result = main.print_report(base_data, test_data)
+        self.assertEqual(1, len(result))
+        self.assertTrue("index2" in result)
+        # Index has conflict in mappings
+        test_data = copy.deepcopy(base_data)
+        del test_data["index1"]["mappings"]["str_key"]
+        self.assertFalse(main.print_report(base_data, test_data))
+
+    def test_validate_plugin_config_unsupported_endpoints(self):
+        with self.assertRaises(ValueError):
+            # No supported endpoints
+            main.validate_plugin_config(BASE_DATA, TEST_KEY)
+
+    def test_validate_plugin_config_missing_host(self):
+        with self.assertRaises(ValueError):
+            # Missing hosts value
+            test_data = create_config_section({})
+            main.validate_plugin_config(test_data, TEST_KEY)
+
+    def test_validate_plugin_config_bad_auth_password(self):
+        with self.assertRaises(ValueError):
+            # User without password
+            test_data = create_config_section(create_plugin_config(["host"], user="test"))
+            main.validate_plugin_config(test_data, TEST_KEY)
+
+    def test_validate_plugin_config_bad_auth_user(self):
+        with self.assertRaises(ValueError):
+            # Password without user
+            test_data = create_config_section(create_plugin_config(["host"], password="test"))
+            main.validate_plugin_config(test_data, TEST_KEY)
+
+    def test_validate_plugin_config_happy_case(self):
+        plugin_config = create_plugin_config(["host"], True, "user", "password")
+        test_data = create_config_section(plugin_config)
+        # Should complete without errors
+        main.validate_plugin_config(test_data, TEST_KEY)
+
+    def test_validate_logstash_config_missing_required_keys(self):
+        with self.assertRaises(ValueError):
+            # Empty input
+            main.validate_logstash_config(dict())
+            # input missing
+            main.validate_logstash_config({"output": ()})
+            # output missing
+            main.validate_logstash_config({"input": ()})
```

Comment:
Same comment here as with the earlier `assertRaises` with multiple test cases -- these should be in separate `with assertRaises` blocks so that 1/ they're all run and 2/ we know if any of them fail.